### PR TITLE
🐛 Fix race condition bug

### LINKED ISSUE #60: 🐛 Initial message is sometimes empty
**Describe the bug**
Sometimes the bot will not send the correct opening message.
image

**Expected behavior**
For this to be created by $AUTHOR in $CHANNEL

**Details**
Happens in V2. No futher investigation was done.

**Additional context**
I have not debugged this and is something I saw while answering a question. View it here:
https://discord.com/channels/574028054238593024/938896754504499293/938896757176295544

### PR DESCRIPTION
**Pull request by MarcusOtter**
Fixes #60 by implementing a `requestId` snowflake (same as `interaction.id` or a newly generated `Snowflake` if we don't have an interaction) to tell different message contexts apart. The ID is deleted from the map of known request IDs once the request is over.

This is a bit messy now (touches on many lines and is passed around a lot), but when we remove message configuration, we won't have to pass this ID around nearly as much (no getMessage() invocation should need it, I think).

### CODE REVIEW

File: src/helpers/messageHelpers.ts [hunk 1]
```diff
@@ -23,22 +23,25 @@ import {
...
-let context: MessageContext = {};
+const contexts: Map<Snowflake, MessageContext> = new Map();
 
 export type MessageKey = keyof NonNullable<NeedleConfig["messages"]>;
 
-export function addMessageContext(additionalContext: Partial<MessageContext>): void {
-	context = Object.assign(context, additionalContext);
+export function addMessageContext(requestId: Snowflake, additionalContext: Partial<MessageContext>): void {
+	const currentContext = contexts.get(requestId);
+	const newContext = Object.assign(currentContext ?? {}, additionalContext);
```
Comment (c43721) [inline src/helpers/messageHelpers.ts:39]: Is there a reason to use object.assign here? Why not do something like `{ ...currentContext, ...additionalContext }`

**Review by c43721 — CHANGES_REQUESTED** [review]

File: src/helpers/messageHelpers.ts (hunk 1 above)
Comment (MarcusOtter) [reply src/helpers/messageHelpers.ts:39]: Mostly cause we use it in other places of the codebase ¯\_(ツ)_/¯
https://github.com/MarcusOtter/discord-needle/blob/main/src/helpers/configHelpers.ts#L36

Comment (MarcusOtter) [reply src/helpers/messageHelpers.ts:39]: Is there a reason to use spreading instead of Object.assign?

Comment (c43721) [reply src/helpers/messageHelpers.ts:39]: ES6 syntax over object IMO. Not sure if the perf. is worth it or not, never tested. Looks cleaner though

Comment (MarcusOtter) [reply src/helpers/messageHelpers.ts:39]: I will leave it for consistency for now. Not sure if it has exact feature parity with Object.assign (I think it does, but maybe there are edge cases where they differ?). Not going to test it out now anyways, can do later if we do an overhaul of style or something.
